### PR TITLE
codeintel: use io.SectionReader for codeintel uploading

### PR DESCRIPTION
**Pull request by Strum355**
Minor tidy up of codeintel upload routine (gzipping, chunking, uploading) to use `io.SectionReader` instead of `io.Reader` factory functions

## Test plan

Covered by existing unit tests

### CODE REVIEW

File: lib/codeintel/upload/upload.go [hunk 1]
```diff
@@ -190,12 +190,12 @@ func uploadMultipartIndexInit(ctx context.Context, httpClient Client, opts Uploa
 }
 
 // uploadMultipartIndexParts uploads the contents available via each of the given reader
```
Comment (efritz) [inline lib/codeintel/upload/upload.go:192]: reader(s) now

File: lib/codeintel/upload/upload.go [hunk 2]
```diff
@@ -277,29 +277,18 @@ func uploadMultipartIndexFinalize(ctx context.Context, httpClient Client, opts U
...
-// looking into a slice of the original reader.
+// splitReader returns a slice of read-seekers into the input ReaderAt, each of max size maxPayloadSize.
 //
-// Two readers from the same factory contain the same content that allows retrying reads from
-// the beginning. The sequential concatenation of each reader produces the content of the original
-// reader.
+// The sequential concatenation of each reader produces the content of the original reader.
 //
-// Each reader is safe to use concurrently. The original reader should be closed when all produced
+// Each reader is safe to use concurrently with others. The original reader should be closed when all produced
```
Comment (efritz) [inline lib/codeintel/upload/upload.go:284]: Love this.

File: lib/codeintel/upload/upload.go (hunk 1 above)
Comment (Strum355) [reply lib/codeintel/upload/upload.go:192]:
```suggestion
// uploadMultipartIndexParts uploads the contents available via each of the given reader(s)
```

File: lib/codeintel/upload/upload.go [hunk 3]
```diff
@@ -189,13 +189,13 @@ func uploadMultipartIndexInit(ctx context.Context, httpClient Client, opts Uploa
 	return id, err
 }
 
-// uploadMultipartIndexParts uploads the contents available via each of the given reader
-// factories to a Sourcegraph instance as part of the same multipart upload as indiciated
+// uploadMultipartIndexParts uploads the contents available via each of the given reader(s)
```
Comment (efritz) [inline lib/codeintel/upload/upload.go:192]:
```suggestion
// uploadMultipartIndexParts uploads the contents available via each of the given readers
```

:laff: the parens were for emphasis

**Review by efritz — APPROVED** [review]

**Pull request merged**